src/sagemaker/pytorch/estimator.py

```diff
@@ -171,7 +172,8 @@ def create_model(
             kwargs["image"] = self.image_name
 
         if "name" not in kwargs:
-            kwargs["name"] = self._current_job_name
+            self._ensure_base_job_name()
+            kwargs["name"] = utils.name_from_base(self.base_job_name)
```

Comment:
I've seen this logic in multiple places. Is it worth creating a re-usable method?